src/Gateways/HeartlandGateway.php

```diff
@@ -210,4 +268,34 @@ public function process_payment( $order_id ) {
 			'redirect' => $is_successful ? $this->get_return_url( $order ) : false,
 		);
 	}
+	
+	private function avsRejectionConditions()
+	{
+	    return array(
+	        'A'  => 'Address matches, zip No Match',
+	        'N'  => 'Neither address or zip code match',
+	        'R'  => 'Retry - system unable to respond',
+	        'S/U'  => 'AVS not supported',
+	        'Z/W'  => '9-digit zip code match, address no match',
+	        'X/Y'  => '5-digit zip code and address match',
```

Comment:
We're only going to get one of these result codes at a time, so we would need these three groups to be split into individual items.

Reply:
ok. updated the conditions